cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraTaskExecutor.java

```diff
@@ -83,49 +76,41 @@ public static final CassandraTaskExecutor parse(
      *
      * @param frameworkId The id of the executor's framework.
      * @param name        The name of the executor.
-     * @param command     The command used to launch the executor.
-     * @param arguments   The arguments passed to the executor.
-     * @param cpus        The cpu shares allocated to the executor.
-     * @param memoryMb    The memory allocated to the executor in Mb.
-     * @param heapMb      The heap allocated to the executor in Mb.
-     * @param apiPort     The port the executor's API will listen on.
-     * @param uris        The URI's for the executor's resources.
-     * @param javaHome    The location of the local java installation for the
-     *                    executor.
      */
     private CassandraTaskExecutor(
         String frameworkId,
         String name,
         String role,
         String principal,
-        String command,
-        List<String> arguments,
-        double cpus,
-        int memoryMb,
-        int heapMb,
-        int apiPort,
-        Set<String> uris,
-        String javaHome) {
-
-        this.info = Protos.ExecutorInfo.newBuilder()
-            .setFrameworkId(Protos.FrameworkID.newBuilder()
+        ExecutorConfig config) {
+
+        Protos.ExecutorInfo.Builder executorBuilder = Protos.ExecutorInfo.newBuilder();
+
+        if ("cni".equalsIgnoreCase(config.getNetworkMode())) {
```

Comment:
It would also be good to put this behind a Capabilities flag:
https://github.com/mesosphere/dcos-commons/blob/20d0e8318451a2ba46609c20d6b8c78693120465/src/main/java/org/apache/mesos/dcos/Capabilities.java#L12